.buildkite/scripts/common.sh

```diff
@@ -52,10 +52,22 @@ retry() {
 
 publish_docker_image() {
     echo "Pushing the docker image "$DOCKER_IMAGE":"$DOCKER_IMAGE_TAG" to the "${DOCKER_REGISTRY}" registry..."
-    docker push "${DOCKER_IMAGE}":"${DOCKER_IMAGE_TAG}"
+    DOCKER_IMAGE=${DOCKER_IMAGE} DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG} make release-docker
 }
 
 docker_logout() {
     echo "Logging out from Docker..."
     docker logout ${DOCKER_REGISTRY}
 }
+
+with_Terraform() {
+    echo "Setting up the Terraform environment..."
+    destFile="terraform.zip"
+    mkdir -p ${WORKSPACE}
+    retry 5 curl -SL -o ${WORKSPACE}/${destFile} "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip"
+    unzip -q ${WORKSPACE}/${destFile} -d ${WORKSPACE}/
+    rm ${WORKSPACE}/${destFile}
+    chmod +x ${WORKSPACE}/terraform
+    terraform version
+    export PATH="${PATH}:${WORKSPACE}"
```

Comment:
is this needed needed if `add_bin_path` is called in the script ?

Reply:
at least for the command `terraform version` in the current view.
if I swap the 71 and 72 lines it can work without add_bin_path

Reply:
fixed